plugins/scaffolder-backend-module-confluence-to-markdowwn/src/actions/confluence/helpers.ts

```diff
@@ -0,0 +1,142 @@
+  self: string;
+}
+
+interface Metadata {
+  mediaType: string;
+}
+
+export interface Result {
+  id: string;
+  type: string;
+  status: string;
+  title: string;
+  metadata: Metadata;
+  _links: Links;
+}
+
+export interface Results {
+  results: Result[];
+}
+
+export const readFileAsString = async (fileDir: string) => {
+  const content = await fs.readFile(fileDir, 'utf-8');
+  return content.toString();
+};
+
+export const parseFromYaml = (content: string) => {
+  return YAML.parse(content);
+};
+
+export const transformToYaml = (content: any) => {
+  return YAML.stringify(content);
+};
+
+export const updateFile = async (content: string, file: string) => {
+  await fs.writeFile(file, content);
+};
+
+export const fetchConfluence = async (relativeUrl: string, config: Config) => {
+  const baseUrl = config.getString('confluence.baseUrl');
+  const token = config.getString('confluence.token');
+  const response: Response = await fetch(`${baseUrl}${relativeUrl}`, {
+    method: 'GET',
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+  if (!response.ok) {
+    throw await ResponseError.fromResponse(response);
+  }
+
+  return response.json() as Promise<any>;
+};
+
+export const getAndWriteAttachments = async (
+  arr: Results,
+  workspace: string,
+  config: Config,
+  mkdocsDir: string,
+) => {
+  const productArr: string[][] = [];
+  const baseUrl = config.getString('confluence.baseUrl');
+  const token = config.getString('confluence.token');
+  await Promise.all(
+    await arr.results.map(async (result: Result) => {
+      const downloadLink = result._links.download;
+      const downloadTitle = result.title.replace(/ /g, '-');
+      if (result.metadata.mediaType !== 'application/gliffy+json') {
+        productArr.push([result.title.replace(/ /g, '%20'), downloadTitle]);
+      }
+
+      const res = await fetch(`${baseUrl}${downloadLink}`, {
+        method: 'GET',
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      // console.log(`${workspace}/${mkdocsDir}docs/img/${downloadTitle}`)
+      if (!res.ok) {
+        throw ResponseError.fromResponse(res);
+      } else if (res.body !== null) {
+        fs.openSync(`${workspace}/${mkdocsDir}docs/img/${downloadTitle}`, 'w');
+        const writeStream = fs.createWriteStream(
+          `${workspace}/${mkdocsDir}docs/img/${downloadTitle}`,
+        );
+        res.body.pipe(writeStream);
+        await new Promise((resolve, reject) => {
+          writeStream.on('finish', () => {
+            resolve(`${workspace}/${mkdocsDir}docs/img/${downloadTitle}`);
+          });
+          writeStream.on('error', reject);
+        });
+      } else {
+        throw new ConflictError(
+          'No Body on the response. Can not save images from Confluence Doc',
+        );
+      }
+    }),
+  );
+  return productArr;
+};
+
+export const createConfluenceVariables = async (url: string) => {
+  let spacekey: string | null = null;
```

Comment:
We avoid using null so I suggest you update these to be `undefined`

Reply:
changed to undefined.